firmware/shared/os/event_flags.h

```diff
@@ -0,0 +1,17 @@
+/// @author Matteo Tullo
+/// @date 2024-02-24
+
+#pragma once
+
+#include <cstdint>
+#include "shared/util/os.h"
+
+namespace shared::osDataType {
+
+class EventFlags {
+public:
+    virtual shared::util::OsStatus Acquire() = 0;
+    virtual shared::util::OsStatus Release() = 0;
```

Comment:
don't need to fully qualify the OsStatus name since we are already in the `shared::util` namespace